lib/nexpose/site.rb

```diff
@@ -430,23 +430,32 @@ def remove_user(user_id)
     end
 
     def self.from_hash(hash)
-      site = new(hash[:name], hash[:scan_template_id])
-      hash.each do |k, v|
-        site.instance_variable_set("@#{k}", v)
-      end
+      hash = Nexpose::SymbolizeHashKeys.symbolize(hash)
+      site = self.json_initializer(hash).deserialize(hash)
 
       # Convert each string address to either a HostName or IPRange object
-      included_scan_targets = { addresses: [], asset_groups: [] }
-      site.included_scan_targets[:addresses].each { |asset| included_scan_targets[:addresses] << HostOrIP.convert(asset) }
-      included_scan_targets[:asset_groups] = site.included_scan_targets[:asset_groups]
-      site.included_scan_targets = included_scan_targets
+      included_addresses = hash[:included_scan_targets][:addresses]
```

Comment:
All of this code was copy/pasted from [``.load``](https://github.com/rapid7/nexpose-client/blob/master/lib/nexpose/site.rb#L504-L534), so it sounds like that was already an issue.

Is the ``:asset_groups`` something I need to look into further, or should we consider this working as intended? 

I can sturdy up the ``:included_scan_targets`` hash.